test/AdaptiveSpinLockTest.cpp

```diff
@@ -62,27 +62,89 @@ TEST(AdaptiveSpinLockTest, WorksWithStdUniqueLock) {
   lock.unlock();
 }
 
-const int thread_num = 7;
-int val;
-pipert::AdaptiveSpinLock* g_lock = nullptr;
-
-void SharedIncrementor(int times) {
-  ASSERT_TRUE(g_lock);
-  for (int i = 0; i < times; i++) {
-    std::lock_guard<pipert::AdaptiveSpinLock> guard(*g_lock);
-    val++;
+namespace {
+
+class SharedIncrementor {
+ public:
+  void StartThreads(void func (SharedIncrementor&)) {
+    for(unsigned int i = 0; i < threads_.size(); ++i)
+      threads_[i] = std::thread(func, std::ref(*this));
+  }
+  void JoinThreads() {
+    for(unsigned int i = 0; i < threads_.size(); ++i)
+      threads_[i].join();
+  }
+
+  void SetThreads(const int count) { threads_.resize(count); }
+  int GetThreads() const { return threads_.size(); }
+  void SetCycles(const int cycles) { cycles_ = cycles; }
+  int GetCycles() const { return cycles_; }
+  void ResetVal() { val_ = 0; }
+  void IncrementVal() { val_++; }
+  int GetVal() const { return val_; }
+
+ private:
+  std::vector<std::thread> threads_;
+  int cycles_;
+  int val_;
+};
+
+void LockEveryStep(SharedIncrementor& shinc) {
+  static pipert::AdaptiveSpinLock slock(2);
+  for(int i = 0; i < shinc.GetCycles(); ++i) {
+    std::lock_guard<pipert::AdaptiveSpinLock> guard(slock);
+    shinc.IncrementVal();
+  }
+}
+void LockYieldEveryStep(SharedIncrementor& shinc) {
+  static pipert::AdaptiveSpinLock slock(2);
```

Comment:
Is there a specific reason for the parameter 2 here? If yes, we should have a test using the default value. Just in case. Maybe these tests could indicate why are they using this value. This is for the user to be able to see the correct usage of this parameter.

Reply:
I still don't like the static initialization of this lock. Why don't we put this into the created object as a class member?

Reply:
I see. The reason for parameters 2 and 1 specifically for the spin cycles was to have at least one spin for the heavily synchronized tests, and to have 1 as the spin value for the last test that tries to force the expiration of the countdown timer with longer operations.

I've put some short comments into the place where these parameters are injected (the helper fixture's ctor, after the adjustment I am currently about to push), to slightly hint at the reason for those specific values.